bastion-launch-gui.sh

```
@@ -0,0 +1,78 @@
+            # Alternative fallback if systemd-run doesn't work
+            if [ $? -ne 0 ]; then
+                echo "systemd-run failed, trying direct launch..."
+    
+                # Get session environment variables
+                display=$(loginctl show-session "$session" -p Display --value 2>/dev/null)
+                wayland_display=$(loginctl show-session "$session" -p XDG_SESSION_ID --value 2>/dev/null | xargs -I{} find /run/user/"$user_uid" -name "wayland-*" -print -quit)
+    
+                if [[ "$session_type" == "wayland" ]] && [ -n "$wayland_display" ]; then
+                    sudo -u "$session_user" env \
+                        XDG_RUNTIME_DIR="/run/user/$user_uid" \
+                        WAYLAND_DISPLAY=$(basename "$wayland_display") \
```

Shellcheck (0.11.0)

[warning] Quote this to prevent word splitting.

(SC2046)